Orm/Xtensive.Orm/Orm/Providers/SqlCompiler.Helpers.cs

```diff
@@ -178,6 +178,16 @@ private static bool IsColumnStub(SqlColumn column)
       return false;
     }
 
+    private static bool IsTypeIdColumn(SqlColumn column)
+    {
+      if (column is SqlUserColumn)
+        return (StringComparer.InvariantCultureIgnoreCase.Compare(column.Name, "TypeId") == 0);

```

Comment:
Why do you use `StringComparer.InvariantCultureIgnoreCase` comparer? Can't it be the `StringComparer.OrdinalIgnoreCase` comparer or just `String.Equals(column.Name, "TypeId", StringComparison.OrdinalIgnoreCase)`? Otherwise this is going to be an expensive check.